src/neuroconv/tools/nwb_helpers.py

```diff
@@ -47,20 +50,35 @@ def get_default_nwbfile_metadata() -> DeepDict:
 
 def make_nwbfile_from_metadata(metadata: dict) -> NWBFile:
     """Make NWBFile from available metadata."""
-    metadata = dict_deep_update(get_default_nwbfile_metadata(), metadata)
+
+    # Validate metadata
+    schema_path = Path(__file__).resolve().parent.parent / "schemas/base_metadata_schema.json"
+    with open(schema_path) as schema_file:
+        base_metadata_schema = json.load(schema_file)
+    try:
+        # jsonschema.validate(metadata, base_metadata_schema)
+        validate_metadata(metadata=metadata, schema=base_metadata_schema)
+    except jsonschema.exceptions.ValidationError as e:
+        print(f"Base metadata is not valid.\n")
+        raise e
+
     nwbfile_kwargs = metadata["NWBFile"]
+    # convert ISO 8601 string to datetime
+    if isinstance(nwbfile_kwargs.get("session_start_time"), str):
+        nwbfile_kwargs["session_start_time"] = datetime.fromisoformat(nwbfile_kwargs["session_start_time"])
```

Comment:
Remind me, PyNWB insists on it being a `datetime` object, right?

Reply:
yes, it does
https://pynwb.readthedocs.io/en/stable/pynwb.file.html#pynwb.file.NWBFile